src/classes/RD2_ScheduleService.cls

```diff
@@ -185,6 +193,128 @@ public without sharing class RD2_ScheduleService {
         return isNeeded;
     }
 
+    /***
+    * @description Calculates the dates of the next n donations
+    * @param rd Recurring Donation record
+    * @param visualizationStartDate Date
+    * @param maxDates Integer
+    * @return List<Installment>
+    */
+    public List<Installment> getNextDonationsForVisualization(npe03__Recurring_Donation__c rd, Date visualizationStartDate, Integer maxDates) {
+
+        if (maxDates == null) {
+            maxDates = DEFAULT_VISUALIZATION_SCHEDULE_COUNT;
+        }
+
+        if (visualizationStartDate == null) {
+            visualizationStartDate = currentDate;
+        }
+
+        Installment ins;
+        List<Installment> vizSchedules = new List<Installment>();
+        currentDate = visualizationStartDate;
+        Integer dateCount = 0;
+        List <RecurringDonationSchedule__c> filteredSchedules;
+
+        List <RecurringDonationSchedule__c> schedules = getExistingSchedules(rd);
+
+        if (schedules == null) {
+            return vizSchedules;
+        }
```

Comment:
@force2b As currently built, getNextDonationsForVisualization() assumes we're dealing with a single RD.  The underlying implementation of getExistingSchedules uses a bulk pattern with an overload for a single RD, so it wouldn't be hard to bulkify getNextDonationsForVisualization().  We'd just need to build a map and iterate through it for the rest of the logic.  I think we should probably make that change.  What do you think?

Reply:
If we make that change, we can use the same overloading pattern used for getExistingSchedules() to make it easy to use for either single RD or bulk and support a return type of map for the bulk pattern and list for the single RD.